### PR TITLE
First crack at building a custom CKAN install

### PR DESCRIPTION
**Pull request by Ooblioob**
Follows the instructions listed in http://docs.ckan.org/en/latest/maintaining/installing/install-from-source.html, CKAN's Official Documentation.

Only notable deviations:
- Installing Python 2.7 instead of using Python 2.6 that it built into CentOS 6.5

### CODE REVIEW

File: provisioners/roles/postgres-ckan/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  yum: name={{item}}
+  with_items: 
+    - postgresql93-devel
+    - postgresql93-server
+    - postgresql93-libs
+    - python-psycopg2
+  sudo: yes
+
+- name: determine if db initialized
+  stat: path=/var/lib/pgsql/9.3/data/pg_hba.conf
```
Comment (mehtadev17) [inline provisioners/roles/postgres-ckan/tasks/main.yml]: You should abstract the path to make it work on servers. 
Path should point towards attached data volume on server.

File: provisioners/roles/postgres-ckan/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  shell: service postgresql-9.3 initdb 
+  when: not db_init.stat.exists
+  register: init_db_result
+  failed_when: "'FAILED' in init_db_result.stderr" 
+  sudo: yes
+
+- name: start postgres db
+  service: name=postgresql-9.3 state=started enabled=yes
+  sudo: yes
+
```
Comment (mehtadev17) [inline provisioners/roles/postgres-ckan/tasks/main.yml]: Split out the following into another role. Split up postgres set up with ckan postgres setup.

File: provisioners/roles/postgres-ckan/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+- name: create datastore db user
+  postgresql_user: 
+    db="{{ckan_datastore_db_name}}" 
+    name="{{ckan_datastore_db_user}}" 
+    password="{{ckan_datastore_db_password}}"
+    #priv="ALL"
+  sudo: yes
+  register: create_datastore_db_user
+  sudo_user: postgres
+
```
Comment (mehtadev17) [inline provisioners/roles/postgres-ckan/tasks/main.yml:47]: Add hostname, password and port options to each of the create dbs.

File: provisioners/roles/ckan/templates/config.ini.j2 [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+ckan.auth.user_delete_organizations = false
+ckan.auth.create_user_via_api = false
+ckan.auth.create_user_via_web = true
+ckan.auth.roles_that_cascade_to_sub_groups = admin
+
+
+## Search Settings
+
+ckan.site_id = {{ckan_site_id}}
+solr_url = http://127.0.0.1:8983/solr
```
Comment (mehtadev17) [inline provisioners/roles/ckan/templates/config.ini.j2]: abstract the solr url so you can specify it.

File: provisioners/roles/ckan/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+
+- name: create crontab for sending activity stream emails
+  cron: name="ckan_send_emails"
+        minute={{ckan_email_flush_interval}}
+        job="echo $(date) >> {{ckan_log_dir}}/email_flush.log && echo '{}' | {{ckan_venv}}/bin/paster post -c {{ckan_config_dir}}/config.ini /api/action/send_email_notifications >> {{ckan_log_dir}}/email_flush.log && echo '' >> {{ckan_log_dir}}/email_flush.log"
+        user="{{ckan_user}}"
+  sudo: yes
+
+# Stop iptables because it blocks some of the necessary ports for
```
Comment (mehtadev17) [inline provisioners/roles/ckan/tasks/main.yml]: Tag this so its not run on production.

**Comment by mehtadev17** [comment]
:+1:

File: provisioners/roles/ckan/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    extra_args="-e"
+
+- name: pip install CKAN requirements
+  pip:
+    requirements="{{ckan_venv}}/src/ckan/requirements.txt"
+    virtualenv="{{ckan_venv}}"
+  environment: 
+    # This is a bit of a hack so that we don't have to spend
+    # time gathering facts. We could just use anisble_env.PATH
+    PATH: /usr/pgsql-9.3/bin:/usr/local/bin:/bin:/usr/bin
```
Comment (mehtadev17) [inline provisioners/roles/ckan/tasks/main.yml:31]: Check with Tijani when installing CKAN.

File: provisioners/roles/postgres-ckan/tasks/main.yml (hunk 1 above)
Comment (marcesher) [inline provisioners/roles/postgres-ckan/tasks/main.yml]: I agree on abstracting, but note that we won't be running any of the postgres bits on servers. This is strictly for Vagrant / local development

File: provisioners/roles/ckan/templates/config.ini.j2 (hunk 4 above)
Comment (marcesher) [inline provisioners/roles/ckan/templates/config.ini.j2]: Yes, we will need that b/c solr will run on a separate server

File: provisioners/roles/postgres-ckan/tasks/main.yml (hunk 1 above)
Comment (Ooblioob) [inline provisioners/roles/postgres-ckan/tasks/main.yml]: I think I'll just split it out into two separate roles as per Dev's suggestion below.

File: provisioners/roles/ckan/tasks/main.yml (hunk 5 above)
Comment (Ooblioob) [inline provisioners/roles/ckan/tasks/main.yml]: I'm planning to just break this off into it's own role so we don't have to fiddle with tags.  That way we can say this is just a local change.

**Comment by marcesher** [comment]
Paired on this, reviewed PR. We've been working with @Ooblioob 's version as the upstream for a while. All's good, time to merge.

**Pull request merged**